src/middleware/authMiddleware.ts

```diff
@@ -5,33 +5,114 @@ import { JwtPayload } from "../types/auth";
 import { User } from "../models/User";
 import { AppDataSource } from "../config/db";
 
-declare global {
-  namespace Express {
-    interface Request {
-      user?: JwtPayload;
-    }
-  }
-}
-
-export const authenticateToken = (
+export const authenticateToken = async (
```

Comment:
rename this function. you are authenticating user not token

Reply:
Oh yes yes, thanks...